src/components/app/details/cicdHistory/TriggerDetails.tsx

```diff
@@ -0,0 +1,295 @@
+)
+
+const Finished = React.memo(
+    ({ status, finishedOn, artifact }: FinishedType) => {
+        return (
+            <div className="flex column left">
+                <div
+                    className={`${status} fs-14 fw-6 ${
+                        TERMINAL_STATUS_COLOR_CLASS_MAP[status.toLowerCase()] || 'cn-5'
+                    }`}
+                >
+                    {status && status.toLowerCase() === 'cancelled' ? 'ABORTED' : status}
+                </div>
+                <div className="flex left">
+                    {finishedOn && finishedOn !== '0001-01-01T00:00:00Z' && (
+                        <time className="cn-7 fs-12 mr-12">
+                            {moment(finishedOn, 'YYYY-MM-DDTHH:mm:ssZ').format(Moment12HourFormat)}
+                        </time>
+                    )}
+                    {artifact && (
+                        <div className="dc__app-commit__hash ">
+                            <img src={docker} className="commit-hash__icon grayscale" />
+                            {artifact.split(':')[1]}
+                        </div>
+                    )}
+                </div>
+            </div>
+        )
+    },
+)
+
+const WorkerStatus = React.memo(
+    ({ message, podStatus, stage }: WorkerStatusType) => {
+        if (!message && !podStatus) return null
+        return (
+            <>
+                <span style={{ height: '80%', borderRight: '1px solid var(--N100)', margin: '0 16px' }} />
+                <div className="flex left column">
+                    <div className="flex left fs-14">
+                        <div className="mr-10">{stage === 'DEPLOY' ? 'Message' : 'Worker'}</div>
+                        {podStatus && (
+                            <div className="fw-6" style={{ color: colorMap[podStatus.toLowerCase()] }}>
+                                {podStatus}
+                            </div>
+                        )}
+                    </div>
+                    {message && <div className="fs-12 cn-7">{message}</div>}
+                </div>
+            </>
+        )
+    },
+)
+
+const ProgressingStatus = React.memo(
+    ({
+        status,
+        message,
+        podStatus,
+        stage,
+        type,
+    }: ProgressingStatusType) => {
+        const [aborting, setAborting] = useState(false)
+        const [abortConfirmation, setAbortConfiguration] = useState(false)
+        const { buildId, triggerId, pipelineId } = useParams<{
+            buildId: string
+            triggerId: string
+            pipelineId: string
+        }>()
+        let abort = null
+        if (type === HistoryComponentType.CI) {
+            abort = () => cancelCiTrigger({ pipelineId, workflowId: buildId })
+        } else if (stage !== 'DEPLOY') {
+            abort = () => cancelPrePostCdTrigger(pipelineId, triggerId)
+        }
+        async function abortRunning() {
+            setAborting(true)
+            const [error] = await asyncWrap(abort())
```

Comment:
Don't we need any fail check around `abort` as it can be null?

Reply:
If it will be null then this line won't get executed anyway

Reply:
Ok